src/x32-transposec/gen/x32-transposec-2x2-rvv.c

```diff
@@ -0,0 +1,106 @@
+// Auto-generated file. Do not edit!
+//   Template: src/x32-transposec/rvv.c.in
+//   Generator: tools/xngen
+//
+// Copyright 2023 SiFive, Inc.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <riscv_vector.h>
+
+#include <assert.h>
+
+#include <xnnpack/common.h>
+#include <xnnpack/math.h>
+#include <xnnpack/transpose.h>
+
+void xnn_x32_transposec_ukernel__2x2_rvv(
+    const uint32_t* input,
+    uint32_t* output,
+    size_t input_stride,
+    size_t output_stride,
+    size_t block_width,
+    size_t block_height,
+    const union xnn_x32_transpose_params params[restrict XNN_MIN_ELEMENTS(1)]) XNN_OOB_READS
+{
+  assert(output_stride >= block_height * sizeof(uint32_t));
+  assert(input_stride >= block_width * sizeof(uint32_t));
+
+  const size_t tile_height = 2;
+  const size_t tile_width = 2;
+  const size_t tile_hbytes = tile_height * sizeof(uint32_t);
+  const size_t tile_wbytes = tile_width * sizeof(uint32_t);
+  const size_t input_reset = tile_wbytes - round_down_po2(block_height, tile_height) * input_stride;
+  const size_t input_offset = tile_height * input_stride;
+  const size_t output_reset = tile_width * output_stride - round_down_po2(block_height, 2) * sizeof(uint32_t);
+
+  const uint32_t* i0 = input;
+
+  uint32_t* o0 = (uint32_t*) output;
+  uint32_t* o1 = (uint32_t*) ((uintptr_t) o0 + output_stride);
+
+  do {
+    size_t bh = block_height;
+    size_t vl = __riscv_vsetvl_e32m1(tile_height);
+    for (; bh >= 2; bh -= 2) {
+        if (block_width >= tile_width) {
+            vuint32m1x2_t tuple = __riscv_vlsseg2e32_v_u32m1x2(i0, input_stride, vl);
```

Comment:
This load is a bit of a bottleneck.  vl is relatively small.   Is there a way a full vector could be loaded?  Or at least 2 tiles, to allow 2 vector units to be more effective.

Reply:
I agree that 2x2 uk is very small; I will update to remove it. We can choose the full vector or the most efficient tile-size ukernel for different hw configs in the [amalgam](https://github.com/google/XNNPACK/tree/master/src/amalgam/gen). Actually, the patch has already been completed. We will include it in the next PR after this one has been merged.